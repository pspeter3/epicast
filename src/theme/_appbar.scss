@import "./animation";
@import "./elevation";
@import "./sizing";
@import "./spacing";
@import "./typography";
@import "./variables";

$appbar__spacing: 8 * $unit;

.appbar {
    @include elevation-medium();
    background-color: $surface;
    color: $white;
    display: flex;
    justify-content: space-between;
    left: 0;
    padding: $subunit;
    position: fixed;
    right: 0;
    top: 0;
}

.appbar__navigation {
    @include sizing-row();
    display: flex;
}

.appbar__title {
    @include title();
    @include spacing-target();
    margin: 0;
}

.appbar__action {
    @include spacing-target();
    border-radius: 50%;
    color: $white;
    transition: background-color $duration-simple $easing-standard,
        color $duration-simple $easing-standard;

    &:focus,
    &:hover {
        background-color: $highlight;
        color: $primary;
    }
}

#root {
    padding: $appbar__spacing $subunit 0 * $unit;
}

@supports (padding: unquote("max(0px)")) {
    .appbar,
    #root {
        padding-left: unquote("max(#{$subunit}, env(safe-area-inset-left))");
        padding-right: unquote("max(#{$subunit}, env(safe-area-inset-right))");
    }

    .appbar {
        padding-top: unquote("max(#{$subunit}, env(safe-area-inset-top))");
    }

    #root {
        // SCSS cannot handle calc and env so we save it as a CSS variable
        --notch: env(safe-area-inset-top);
        padding-top: unquote("calc(#{$appbar__spacing} + var(--notch))");
        padding-bottom: unquote("env(safe-area-inset-bottom)");
    }
}
