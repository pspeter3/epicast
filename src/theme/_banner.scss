@import "./elevation";
@import "./spacing";
@import "./typography";
@import "./variables";

.banner {
    @include elevation-small();
    @include caption();
    @include spacing-y();
    background-color: $primary;
    border-radius: $border-radius;
    color: $white;
    text-align: center;
}
