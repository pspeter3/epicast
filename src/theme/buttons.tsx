import * as React from "react";

export const IconButton: React.SFC<React.ButtonHTMLAttributes<HTMLButtonElement>> = props => {
    return <button className="icon-button" type="button" {...props} />;
};

export const SubHeaderButton: React.SFC<React.ButtonHTMLAttributes<HTMLButtonElement>> = props => {
    return <button className="subheader-button" type="button" {...props} />;
};

export const NumericSubHeaderButton: React.SFC<
    React.ButtonHTMLAttributes<HTMLButtonElement>
> = props => {
    return <button className="subheader-button numeric" type="button" {...props} />;
};

export const PrimarySubHeaderButton: React.SFC<
    React.ButtonHTMLAttributes<HTMLButtonElement>
> = props => {
    return <button className="subheader-button primary" type="button" {...props} />;
};

export const BottomButton: React.SFC<React.ButtonHTMLAttributes<HTMLButtonElement>> = props => {
    return <button className="bottom-button" type="button" {...props} />;
};
