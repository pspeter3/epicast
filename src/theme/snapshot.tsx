import * as React from "react";
import TestRenderer from "react-test-renderer";

export const snapshot = (name: string, SFC: React.SFC<{}>): void => {
    describe(name, () => {
        it(`should match snapshot`, () => {
            const component = TestRenderer.create(<SFC>Snapshot</SFC>);
            const tree = component.toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
};
