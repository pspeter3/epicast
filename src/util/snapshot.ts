import { shallow } from "enzyme";
import * as React from "react";

export type Factory = () => React.ReactElement<any>;

export const snapshot = (factory: Factory): (() => void) => {
    return () => {
        expect(shallow(factory())).toMatchSnapshot();
    };
};

export const snapshotSuite = (name: string, factory: Factory): void => {
    describe(name, () => {
        it("should match snapshot", snapshot(factory));
    });
};
