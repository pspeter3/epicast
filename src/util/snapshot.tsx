import * as React from "react";
import TestRenderer from "react-test-renderer";

export type Factory = () => React.ReactElement<any>;

export const snapshot = (factory: Factory): (() => void) => {
    return () => {
        const component = TestRenderer.create(factory());
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    };
};

export const snapshotSuite = (name: string, factory: Factory): void => {
    describe(name, () => {
        it("should match snapshot", snapshot(factory));
    });
};
