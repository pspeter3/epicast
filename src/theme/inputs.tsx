import * as React from "react";
import { IconButton } from "./buttons";
import { CheckCircleIcon, CircleIcon } from "./icons";

export const NumericInput: React.SFC<React.InputHTMLAttributes<HTMLInputElement>> = props => {
    return <input className="numeric-input" type="number" {...props} />;
};

export const Checkbox: React.SFC<React.ButtonHTMLAttributes<HTMLButtonElement>> = props => {
    const checked = props["aria-checked"];
    return (
        <IconButton role="checkbox" aria-checked={checked} {...props}>
            {checked ? <CheckCircleIcon /> : <CircleIcon />}
        </IconButton>
    );
};
