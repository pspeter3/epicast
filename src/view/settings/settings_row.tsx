import * as React from "react";
import { IconButton } from "../../theme/buttons";
import { MinusCircleIcon, PlusCircleIcon } from "../../theme/icons";
import { NumericInput } from "../../theme/inputs";
import { Row } from "../../theme/layout";
import { Text } from "../../theme/typography";

export interface Props {
    name: string;
    displayName?: string;
    value: number;
    onDecrement: (name: string) => void;
    onChange: (name: string, value: number) => void;
    onIncrement: (name: string) => void;
}

export class SettingsRow extends React.PureComponent<Props, {}> {
    public render() {
        const { name, displayName, value } = this.props;
        return (
            <Row>
                <Text>{displayName || name}</Text>
                <IconButton onClick={this._onDecrement}>
                    <MinusCircleIcon />
                </IconButton>
                <NumericInput onChange={this._onChange} value={value} />
                <IconButton onClick={this._onIncrement}>
                    <PlusCircleIcon />
                </IconButton>
            </Row>
        );
    }

    private _onDecrement = () => {
        this.props.onDecrement(this.props.name);
    };

    private _onChange = (evt: React.ChangeEvent<HTMLInputElement>) => {
        const value = evt.target.value;
        if (value) {
            this.props.onChange(this.props.name, parseInt(evt.target.value, 10));
        }
    };

    private _onIncrement = () => {
        this.props.onIncrement(this.props.name);
    };
}
