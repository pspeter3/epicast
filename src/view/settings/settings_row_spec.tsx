import { shallow } from "enzyme";
import * as React from "react";
import { IconButton } from "../../theme/buttons";
import { NumericInput } from "../../theme/inputs";
import { snapshot } from "../../util/snapshot";
import { SettingsRow } from "./settings_row";

describe("SettingsRow", () => {
    it(
        "should match snapshot with name",
        snapshot(() => (
            <SettingsRow
                name="name"
                value={1}
                onDecrement={jest.fn()}
                onChange={jest.fn()}
                onIncrement={jest.fn()}
            />
        )),
    );

    it(
        "should match snapshot with displayName",
        snapshot(() => (
            <SettingsRow
                name="name"
                displayName="Display Name"
                value={1}
                onDecrement={jest.fn()}
                onChange={jest.fn()}
                onIncrement={jest.fn()}
            />
        )),
    );

    it("should handle decrement", () => {
        const name = "name";
        const onDecrement = jest.fn();
        const tree = shallow(
            <SettingsRow
                name={name}
                value={1}
                onDecrement={onDecrement}
                onChange={jest.fn()}
                onIncrement={jest.fn()}
            />,
        );
        tree
            .find(IconButton)
            .first()
            .simulate("click");
        expect(onDecrement).toHaveBeenCalledWith(name);
    });

    it("should handle change", () => {
        const name = "name";
        const value = 3;
        const onChange = jest.fn();
        const tree = shallow(
            <SettingsRow
                name={name}
                value={1}
                onDecrement={jest.fn()}
                onChange={onChange}
                onIncrement={jest.fn()}
            />,
        );
        tree.find(NumericInput).simulate("change", { target: { value: value.toString() } });
        expect(onChange).toHaveBeenCalledWith(name, value);
    });

    it("should handle empthy change", () => {
        const name = "name";
        const onChange = jest.fn();
        const tree = shallow(
            <SettingsRow
                name={name}
                value={1}
                onDecrement={jest.fn()}
                onChange={onChange}
                onIncrement={jest.fn()}
            />,
        );
        tree.find(NumericInput).simulate("change", { target: { value: "" } });
        expect(onChange).not.toHaveBeenCalled();
    });

    it("should handle increment", () => {
        const name = "name";
        const onIncrement = jest.fn();
        const tree = shallow(
            <SettingsRow
                name={name}
                value={1}
                onDecrement={jest.fn()}
                onChange={jest.fn()}
                onIncrement={onIncrement}
            />,
        );
        tree
            .find(IconButton)
            .last()
            .simulate("click");
        expect(onIncrement).toHaveBeenCalledWith(name);
    });
});
