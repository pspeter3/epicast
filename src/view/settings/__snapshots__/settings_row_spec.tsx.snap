// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsRow should match snapshot with displayName 1`] = `
<Component>
  <Component>
    Display Name
  </Component>
  <Component
    onClick={[Function]}
  >
    <Component />
  </Component>
  <Component
    onChange={[Function]}
    value={1}
  />
  <Component
    onClick={[Function]}
  >
    <Component />
  </Component>
</Component>
`;

exports[`SettingsRow should match snapshot with name 1`] = `
<Component>
  <Component>
    name
  </Component>
  <Component
    onClick={[Function]}
  >
    <Component />
  </Component>
  <Component
    onChange={[Function]}
    value={1}
  />
  <Component
    onClick={[Function]}
  >
    <Component />
  </Component>
</Component>
`;
